mlrun/utils/notifications/notification/mail.py

```diff
@@ -84,13 +84,28 @@ def enrich_default_params(
         params.setdefault("server_port", DEFAULT_SMTP_PORT)
 
         default_mail_address = params.pop("default_email_addresses", "")
-        email_addresses = params.get("email_addresses", default_mail_address)
-        if isinstance(email_addresses, list):
-            email_addresses = ",".join(email_addresses)
-        params["email_addresses"] = email_addresses
+        params["email_addresses"] = cls._merge_mail_addresses(
+            default_mail_address, params.get("email_addresses", "")
+        )
 
         return params
 
+    @classmethod
+    def _merge_mail_addresses(
+        cls,
+        default_mail_address: typing.Union[str, list],
+        email_addresses: typing.Union[str, list],
+    ):
+        if isinstance(default_mail_address, str):
+            default_mail_address = (
+                default_mail_address.split(",") if default_mail_address else []
+            )
+        if isinstance(email_addresses, str):
+            email_addresses = email_addresses.split(",") if email_addresses else []
+        email_addresses.extend(default_mail_address)
```

Comment:
I think you should remove any duplicate emails if they exist

Reply:
But why do you need the same recipient email in the list?

Reply:
Because we want it to be real email

Reply:
I agree with Moran here, it makes no sense to have the same recipient more than once.
You can use a `set()` to avoid duplications

Reply:
This logic shouldn't be on the "client" side, the smtp server should do it.
See for example this:
![image](https://github.com/user-attachments/assets/2ca71cef-1140-4432-86af-4a2da00c05ea)

Here I specified your mail 3 times.
Only after sending we can see your mail shown only once:
![image](https://github.com/user-attachments/assets/50c3fe62-47b9-44f5-a880-37478f4dec89)

But as client I mentioned your mail 3 times, the remove duplication should be on the smtp server (because for example if there are groups, this logic is not simple as use `set` for example)